### PR TITLE
Use SWR to get the checkin numbers- no CSS

### LINKED ISSUE #3: Fetch Data & Stream
Next is magic. A wonderful library called `SWR` allows you to poll a backend, implementing automatic caching and polling.

https://swr.vercel.app

Your task:
1. Define an API endpoint on your Next server
2. This API endpoint should query SE API backend for # checkins.
3. Display count of checkins returned.

### PR DESCRIPTION
**Pull request by masnoble**
### Summary

This commit includes the changes to the next-auth, the index.tsx and other config files to enable the use of SWR to retrieve the checkin numbers from the SE-API.

### Testing

Through testing the web app, the app can successfully get the numbers. The time interval is set to 2000 hours so that a user may actually get some sort of results. 

###### notes on testing:
There were some cases where the app would send an invalid token to the API. This seemed to be caused by the session having an "authenticated" status even when the user did not yet have a chance to sign in. I believe that this was resolved, but I wanted to get this PR up before the sprint, and I did not have time to verify it completely

### Future work

* We would want to split this into a separate page- This doesn't belong in the index.tsx. I feel there should be some sort of dashboard page and a designated server-side api that can possibly do the fetching rather than having a fat client-side

* Identify some aspects that could be split off into SSR- Like said before, this client-side could get very fat if too much computation is done on the client side. If we could handle all of the polling from the server and just use SWR from the client to query that new server value, that might be ideal. This could help the client by using the server to manage the ID_token as well and it may improve security by moving the id_token sharing away from the client side where it could be viewed on the GET requests.

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
   reactStrictMode: true,
+  env: {
+    VITE_BACKEND: process.env.VITE_BACKEND,
```
Comment (SpBills) [inline next.config.js:5]: Great job w/ this

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -1,6 +1,11 @@
 import '@/styles/globals.css'
 import type { AppProps } from 'next/app'
+import { SessionProvider } from 'next-auth/react'
 
 export default function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  return (
+    <SessionProvider session={pageProps.session}>
```
Comment (SpBills) [inline src/pages/_app.tsx:7]: Cool. Didn't know this was necessary. Looking forward to learning about this.

File: src/pages/index.tsx [hunk 3]
```diff
@@ -1,17 +1,54 @@
...
+      const number_of_checkins = Array.isArray(all_check_ins) ? all_check_ins.filter((checkin) => (Date.now() - new Date(checkin.created).getTime()) < lim): [];
+      setCheckins(number_of_checkins.length);
+    }
+  }
+
+  if (status === "authenticated") {
+      return (
+        <div>
+          <div>
+            <div>checkins in {limit} hours: {checkins ?? 0}</div>
```
Comment (SpBills) [inline src/pages/index.tsx]: Eventually should unnest these divs lol.

File: src/pages/index.tsx [hunk 4]
```diff
@@ -1,17 +1,54 @@
...
-      </main>
-    </>
-  )
-}
+
+  const { data: session, status } = useSession()
+
+  useEffect(() => {
+    header = new Headers({ "Authorization": `Bearer ${session?.id_token}` })
+    fetcher = (url: string) => fetch(url, { headers: header }).then((res) => res.json());
```
Comment (SpBills) [inline src/pages/index.tsx]: Hm. I dislike the fetcher being stateful. Instead, try and make it a function that takes the header as a param. SWR should have a way of setting headers in the fetch.

**Review by SpBills — CHANGES_REQUESTED** [review]
Just an SWR change. Great work overall.

File: src/pages/index.tsx (hunk 3 above)
Comment (masnoble) [reply src/pages/index.tsx]: lol yeah--- I'll change this with the swr changes... accidentally left those in there from a previous template thingy

File: src/pages/index.tsx (hunk 4 above)
Comment (masnoble) [reply src/pages/index.tsx]: This was addressed in new commit. please see if this is better

File: src/pages/index.tsx [hunk 5]
```diff
@@ -3,24 +3,22 @@ import useSWR from "swr";
 import { signIn, useSession } from "next-auth/react"
 import { ICheckIn } from "@/models/checkins";
 
-let header = new Headers({ "Authorization": `Bearer placeholder-will-be-set` })
-let fetcher = (url: string) => fetch(url, { headers: header }).then((res) => res.json());
+const fetcher = (params: [string, string]) =>
+  fetch(params[0], { headers: { 'Authorization': 'Bearer ' + params[1]} } )
```
Comment (SpBills) [inline src/pages/index.tsx:7]: man actually knows functional programming. let's go dude.

**Review by SpBills — APPROVED** [review]
LGTM

**Pull request merged**